extensions/amp-sidebar/1.0/base-element.js

```diff
@@ -33,10 +36,37 @@ export class BaseElement extends PreactBaseElement {
 
   /** @override */
   init() {
+    //console.log(this.element);
+    const realChildren = childNodes(this.element, () => true);
+
+    //console.log(realChildren);
+
+    const children = realChildren.map((child) => {
+      if (
+        child.nodeName === 'NAV' &&
+        child.hasAttribute('toolbar') &&
+        child.hasAttribute('toolbar-target')
+      ) {
+        const toolbarShim = memo(child, 'toolbar', bindToolbarShimToElement);
+
+        return (
+          <SidebarToolbar
+            as={toolbarShim}
+            toolbar={child.getAttribute('toolbar')}
+            toolbarTarget={child.getAttribute('toolbar-target')}
+          >
+            {child.children}
```

Comment:
Not sure if this is the correct format for passing children?

Reply:
What are the `child.children` in this case? Are these `HTMLCollection<!Element>`? If so, could you explain how this ends up where you want it to? Does it get cloned on a way? Converted to VNodes?